src/engine/x86-64/X86_64Interpreter.v3

```diff
@@ -1302,6 +1303,39 @@ class X86_64InterpreterGen(ic: X86_64InterpreterCode, w: DataWriter) {
 		bindHandler(Opcode.I64_STORE16);
 		genStore(asm.movw_m_r);
 	}
+	def genAtomicLoad(a: X86_64Gpr, b: X86_64Addr) -> this {
```

Comment:
This generates the same code for every atomic load and store, regardless of the size. There should be different instructions in for 8, 16, and 32 bit loads. Also, it looks like loads actually do store to memory via the `xchg`. For example, see the V8 implementation here: https://github.com/v8/v8/blob/main/src/wasm/baseline/x64/liftoff-assembler-x64-inl.h#L544

Reply:
I see. It appears my assumption was wrong for xchg_m_r  for loading as well. I will address them right away.